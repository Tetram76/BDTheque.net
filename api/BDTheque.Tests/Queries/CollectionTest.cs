namespace BDTheque.Tests.Queries;

using Xunit.Abstractions;

public class CollectionTest(ITestOutputHelper testOutputHelper) : DatabaseTest(testOutputHelper)
{
    [Fact]
    public async Task GetCollectionsTest()
    {
        Editeur[] editeurs =
        [
            new Editeur
            {
                Nom = "E_Aa"
            },
            new Editeur
            {
                Nom = "E_Bb"
            }
        ];
        Collection[] collections =
        [
            new Collection
            {
                Nom = "Aa",
                Editeur = editeurs[0]
            },
            new Collection
            {
                Nom = "Bb",
                Editeur = editeurs[1]
            },
            new Collection
            {
                Nom = "Cc",
                Editeur = editeurs[0]
            }
        ];

        RegisterFinalizer(() => RemoveEntities(editeurs));
        await AddEntities(editeurs);

        RegisterFinalizer(() => RemoveEntities(collections));
        await AddEntities(collections);

        string response = await TestServices.ExecuteRequestAsync(
            b => b.SetQuery(
                """
                query {
                    collectionList(last: 10, where: {initiale: {eq: "A"}}, order: {nom: ASC}) {
                      pageInfo {
                        hasNextPage
                        hasPreviousPage
                      }
                      edges {
                        cursor
                        node {
                          id
                          nom
                        }
                      }
                    }
                }
                """
            )
        );

        response.Should().MatchSnapshot(options => options.IgnoreAllFields("id"));
    }

    [Fact]
    public async Task GetCollectionsByEditeurTest()
    {
        Editeur[] editeurs =
        [
            new Editeur
            {
                Nom = "Aa"
            },
            new Editeur
            {
                Nom = "Bb"
            }
        ];
        Collection[] collections =
        [
            new Collection
            {
                Nom = "Aa",
                Editeur = editeurs[0]
            },
            new Collection
            {
                Nom = "Bb",
                Editeur = editeurs[1]
            },
            new Collection
            {
                Nom = "Cc",
                Editeur = editeurs[0]
            }
        ];

        RegisterFinalizer(() => RemoveEntities(editeurs));
        await AddEntities(editeurs);

        RegisterFinalizer(() => RemoveEntities(collections));
        await AddEntities(collections);

        string response = await TestServices.ExecuteRequestAsync(
            b => b.SetQuery(
                """
                query {
                    collectionList(last: 10, where: {editeur: {initiale: {eq: "A"}}}, order: {nom: ASC}) {
                      pageInfo {
                        hasNextPage
                        hasPreviousPage
                      }
                      edges {
                        cursor
                        node {
                          id
                          nom
                        }
                      }
                    }
                }
                """
            )
        );

        response.Should().MatchSnapshot(options => options.IgnoreAllFields("id"));
    }
}
