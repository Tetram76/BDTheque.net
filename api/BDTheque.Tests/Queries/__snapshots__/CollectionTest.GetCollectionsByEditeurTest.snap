{
  "data": {
    "collectionList": {
      "pageInfo": {
        "hasNextPage": false,
        "hasPreviousPage": false
      },
      "edges": [
        {
          "cursor": "MA==",
          "node": {
            "id": "732c33747be64d25a95762f503f37afa",
            "nom": "Aa"
          }
        },
        {
          "cursor": "MQ==",
          "node": {
            "id": "8497e95245da452993abaee7b8ba7286",
            "nom": "Cc"
          }
        }
      ]
    }
  }
}
